src/rpc.rs

```diff
@@ -0,0 +1,489 @@
+                        }
+                    }
+                }
+                Err(RecvError::Lagged(count)) => {
+                    error!("Missed {} updates", count);
+                    // Decide how to handle missed updates
+                }
+                Err(RecvError::Closed) => {
+                    error!("Update channel closed");
+                    break;
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+
+    async fn handle_app_call(&self, app_call: VeilidAppCall) -> Result<()> {
+        let call_id = app_call.id();
+        let message = app_call.message();
+    
+        // Attempt to parse the message with Protocol Buffers
+        let app_call_request = AppCallRequest::decode(message);
+        let response = match app_call_request {
+            Ok(request) => {
+                info!("Received AppCall with command: {}", request.command);
+                self.process_app_call(request).await
+            }
+            Err(err) => {
+                error!("Failed to parse AppCall message: {}", err);
+                Ok(AppCallResponse {
+                    success: false,
+                    message: "Invalid AppCall message format".to_string(),
+                })
+            }
+        };
+    
+        // Encode the AppCallResponse
+        let mut buf = Vec::new();
+        response?.encode(&mut buf).map_err(|e| {
+            error!("Failed to encode AppCallResponse: {}", e);
+            anyhow!("Failed to encode AppCallResponse: {}", e)
+        })?;
+    
+        // Send the response using VeilidAPI's app_call_reply
+        self.backend
+            .get_veilid_api()
+            .await
+            .ok_or_else(|| anyhow!("Veilid API not available"))?
+            .app_call_reply(call_id, buf)
+            .await
+            .map_err(|e| {
+                error!("Failed to send AppCall reply: {}", e);
+                anyhow!("Failed to send AppCall reply: {}", e)
+            })?;
+    
+        Ok(())
+    }
+    
+
+    /// Process the AppCallRequest and generate a response
+    async fn process_app_call(&self, request: AppCallRequest) -> Result<AppCallResponse, anyhow::Error> {
+        match request.command.as_str() {
+            "ping" => Ok(AppCallResponse {
+                success: true,
+                message: "pong".to_string(),
+            }),
+            "echo" => Ok(AppCallResponse {
+                success: true,
+                message: request.payload,
+            }),
+            _ => Ok(AppCallResponse {
+                success: false,
+                message: format!("Unknown command: {}", request.command),
+            }),
+        }
+    }
+}
+
+
+pub async fn start_rpc_server(backend: Backend, addr: &str) -> Result<()> {
+    // Path to the keys file
+    let base_dir = std::env::current_dir().expect("Failed to get current directory");
+    let keys_path = base_dir.join("group_keys.txt");
```

Comment:
Instead we should have the existing code for tracking joined groups in the backend. RPC service should init the backend like normal and then join groups via RPC calls. It should never create its own groups, only join groups created in the app.

Reply:
I switched to using `load_known_groups`
start_rpc_server no longer creates a group 
it will load know groups or look for a group_keys.txt
I dont think the group_keys.txt is correct and will need to be refactored

Reply:
f4f56b0639c67c350b185bf1556490a34f7288f8

Reply:
Lets remove group_keys.txt and only load groups via the AppCalls to join group at runtime + the backend load_groups_from_disk API on boot

Reply:
#18 -> Removes the `.txt` that used to load group keys